### PR TITLE
Adding the moorres voting algorithm in the arrays folder

### PR DESCRIPTION
**Pull request by satyam4484**
## Related Issue
- moore's voting algorithm #47

## Proposed Changes
- Added the Algorithm 

## Additional Info
- Any additional information or context

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the documentation.
- [ ] 🎀 I have updated the documentation accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by Lakhankumawat** [comment]
A mentor will review this pull request shortly , please have patience.

**Comment by Lakhankumawat** [comment]
Hey @satyam4484  , make sure your Pull Request is descriptive and complete the checklist .
I would recommend you to go through this tutorial video of the project [here](https://youtu.be/kIFwqp9B7pQ) ,thanks.

For contribution and more guidelines i would recommend you to go through this 
tutorial video of the project [here](https://youtu.be/7UgC3HDETRA), thanks.

**Comment by satyam4484** [comment]
could you tell me what to do in readme file of particular folder

**Comment by satyam4484** [comment]
also what to do in last three check boxes

**Comment by Lakhankumawat** [comment]
> could you tell me what to do in readme file of particular folder

Please have a look a contribution guidelines in the main Repository of the project , there it is mentioned .

**Comment by Lakhankumawat** [comment]
> also what to do in last three check boxes

just ignore them

**Comment by satyam4484** [comment]
Please have a look a contribution guidelines in the main Repository of the project , there it is mentioned . 
there is nothing about readme files could you tell me where it is

**Comment by n4i9kita** [comment]
@satyam4484 , [here](https://github.com/Lakhankumawat/LearnCPP/tree/main/S-SortingAlgorithms)
If you can write properly, then only take up the task of writing readme.

**Comment by satyam4484** [comment]
yes i can do that

**Comment by satyam4484** [comment]
could help me in that part

File: A-Array/MooresVotingAlogorithm.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+};
+
+int main ( ) {
+    vector<int>arr{2,2,1,1,1,2,2};
+    Solution s;
+    cout<<s.majorityElement(arr);
+    return 0;
+}
+
+// Time Complexcity -   O ( N ) as we are doing a single traversal on elements
```
Comment (n4i9kita) [inline A-Array/MooresVotingAlogorithm.cpp:43]: Time complexity and space complexity will be mentioned in the readme, and can be removed from here.

File: A-Array/MooresVotingAlogorithm.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    int majorityElement(vector<int>& nums) {
+        /*  
+        Intution : If there exists a element which is repeated more than n/2 time 
+                   then there are only n/2-1 elements left which can be removed by n/2 elements 
+                   and the element which remains will be our ans .
+        */
+        int countmajority = 0 , majorelement = 0;  // will count of the number of majority element and will also store which one is the majority element 
+        int n = nums.size();
+        for(int i=0;i<n;i++) {
+            // if we not found any majority element then who ever comes first will be our majority element
```
Comment (n4i9kita) [inline A-Array/MooresVotingAlogorithm.cpp:17]: Write shorter comments if you want to write in-line comments. 
__The code must speak for itself for the most part.__

File: A-Array/MooresVotingAlogorithm.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
+// problem link --> https://leetcode.com/problems/majority-element/
+
+#include<bits/stdc++.h>
```
Comment (n4i9kita) [inline A-Array/MooresVotingAlogorithm.cpp:3]: Use appropriate header files. 
Use vector, iostream etc to be more precise with the libraries we use.

File: A-Array/MooresVotingAlogorithm.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            
+        }
+        // finally the majorelement will store the majority element as it is given there exists majority element
+        return majorelement;
+        
+    }
+};
+
+int main ( ) {
+    vector<int>arr{2,2,1,1,1,2,2};
```
Comment (n4i9kita) [inline A-Array/MooresVotingAlogorithm.cpp:37]: Input the array from  user, rather than defining it yourself.

**Comment by Lakhankumawat** [comment]
No response from the contributor ??

**Comment by satyam4484** [comment]
sorry, Lakhan currently I am in hospital due to some health issues, that's why unable to work as per the requirement. I will get discharged next Sunday, till then I will not be able to make a contribution.
Could you wait for the next Sunday else you can assign it someone else will make a new contribution if someone will do these
Thank You

**Comment by Lakhankumawat** [comment]
Don't worry , get well soon and this PR won't be assigned to anyone else , you can take your time and do it , I'll be glad to merge it at last , do the required changes when you are free .

**Comment by n4i9kita** [comment]
@satyam4484 
Wishing you a speedy recovery.
We understand that you are in the hospital, but if there is a delay in the future, make sure to let the mentors or admin know before they have to ask you.

**Comment by satyam4484** [comment]
i have done the changes suggested by you , but how should I show them to here can you suggest me

File: A-Array/MooresVotingAlogorithm.cpp [hunk 5]
```diff
@@ -1,44 +1,48 @@
...
         int n = nums.size();
         for(int i=0;i<n;i++) {
-            // if we not found any majority element then who ever comes first will be our majority element
+            // get the majority element if not exist
             if(countmajority == 0) {
                 majorelement = nums[i];
             }
 
-            // if the current is same as our majority element then increase its count 
+            // if it is the majority count it
```
Comment (n4i9kita) [inline A-Array/MooresVotingAlogorithm.cpp:24]: I think the comment is not clear.

File: A-Array/MooresVotingAlogorithm.cpp [hunk 6]
```diff
@@ -1,44 +1,48 @@
...
 public:
     int majorityElement(vector<int>& nums) {
-        /*  
-        Intution : If there exists a element which is repeated more than n/2 time 
-                   then there are only n/2-1 elements left which can be removed by n/2 elements 
-                   and the element which remains will be our ans .
-        */
-        int countmajority = 0 , majorelement = 0;  // will count of the number of majority element and will also store which one is the majority element 
+         // count of majority element and the majority element
+        int countmajority = 0 , majorelement = 0;
```
Comment (n4i9kita) [inline A-Array/MooresVotingAlogorithm.cpp:16]: Is your code able to handle the case when there is no majority element?

File: A-Array/MooresVotingAlogorithm.cpp (hunk 5 above)
Comment (satyam4484) [reply A-Array/MooresVotingAlogorithm.cpp:24]: i have change the things you are looking the first commit , check the next one

File: A-Array/MooresVotingAlogorithm.cpp (hunk 6 above)
Comment (satyam4484) [reply A-Array/MooresVotingAlogorithm.cpp:25]: I have added it, can you pls check the last commit

**Comment by n4i9kita** [comment]
LGTM.

**Comment by n4i9kita** [comment]
Do you want to write the Readme for this algorithm as well?

**Comment by satyam4484** [comment]
Actually, i don't know much about writing the readme content.

**Comment by Lakhankumawat** [comment]
You need to follow particular guidelines for your README , please follow
these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) 
also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) 
for more please go through these [here](https://youtu.be/7UgC3HDETRA)

**Comment by Lakhankumawat** [comment]
Practice on this https://dillinger.io/

**Comment by n4i9kita** [comment]
@Lakhankumawat he has not written readme. Please look into it regarding what needs to be done.

**Comment by aditi944** [comment]
closed

**Pull request closed without merging**